### ISSUE TITLE
training question on LLama_factory

### ISSUE THREAD

**Issue by Haozon** [astramind-ai/Mixture-of-depths#2]
Hello, you have provided a great job. But I myself keep having gradient explosion problems when inserting it into llama_factory training, do you know why?
Among them I have modified the code as follows:
![image](https://github.com/astramind-ai/Mixture-of-depths/assets/74298632/bdbf36b0-e5ac-49f4-b523-a604ff19d727)

These places may need to be modified, otherwise they cannot be trained because the length of attention_mask does not match.

And I added this to the main training code
![image](https://github.com/astramind-ai/Mixture-of-depths/assets/74298632/4ab616a5-5e20-417d-be25-ed5b20b5d57d)
Have you been successful in your training? What configuration does the training use?

The terminal print result is
![image](https://github.com/astramind-ai/Mixture-of-depths/assets/74298632/f6a5fb3a-0d20-47fd-bd7d-a5c0755f3f5e)

**Comment by Haozon** [comment]
I soved it but I am confused

**Comment by FB-wh** [comment]
I have the same problem. How did you solve it?
